hack/build.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+set -ex
+
+go version
+
+GIT_VERSION=$(git describe --always --tags || true)
+VERSION=${CI_VERSION:-${GIT_VERSION}}
+GIT_COMMIT=$(git rev-list -1 HEAD || true)
+COMMIT=${CI_COMMIT:-${GIT_COMMIT}}
+BUILD_DATE=$(date --utc -Iseconds)
+
+mkdir -p bin
+
+LDFLAGS="-s -w "
+LDFLAGS+="-X github.com/medik8s/node-healthcheck-operator/version.Version=${VERSION} "
+LDFLAGS+="-X github.com/medik8s/node-healthcheck-operator/version.GitCommit=${COMMIT} "
+LDFLAGS+="-X github.com/medik8s/node-healthcheck-operator/version.BuildDate=${BUILD_DATE} "
+GOFLAGS=-mod=vendor CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="${LDFLAGS}" -o bin/manager main.go
```

Comment:
Building the operator container is failing since the new build command include `GOFLAGS=-mod=vendor` while there is an inconsistency of Golang version between [Dockerfile](https://github.com/medik8s/fence-agents-remediation/blob/main/Dockerfile#L6) and [go.mod](https://github.com/medik8s/fence-agents-remediation/blob/main/go.mod#L3).

IIUC removing the above flag will allow the build to finish successfully, but it would makes more sense to align go.mod and Dockerfile